src/librustdoc/test.rs

```diff
@@ -346,6 +391,17 @@ fn run_test(test: &str, cratename: &str, filename: &FileName, line: usize,
     };
     cmd.env(var, &newpath);
 
+    if persist_executable {
+        let mut target_path = target_path.unwrap();
+        target_path.push("path.txt");
+
+        std::fs::write(
+            target_path,
+            newpath.as_os_str().to_str().unwrap().as_bytes(),
```

Comment:
`Path` has a `to_str` method, you souldn't need to go through `OsStr`

https://doc.rust-lang.org/nightly/std/path/struct.Path.html#method.to_str

Reply:
`newpath` is an `OsString`

Reply:
Ah, well `OsString` coerces to an `OsStr`, so you should be able to call `to_str()` directly on the `OsString`, but not a big deal